.github/actions/parse-version/script.php

```diff
@@ -70,13 +70,30 @@ function output(string $name, string $value) {
     // If a fixed support range is set, use this.
     if (!empty($plugin->supported)) {
         [$lower, $upper] = $plugin->supported;
+        // If within permitted ranges, e.g. 36 is between [35, 39], include it.
         if ($lower <= $coreVersion && $coreVersion <= $upper) {
             return true;
         }
-        // For non master branches, immediately return false as they should not be included in the run.
-        if ($entry['moodle-branch'] !== 'master') {
-            return false;
+
+        // If this iteration is on master, check if the upper supported value implies master should be tested or not.
+        // e.g. for [35, 500], 500 doesn't exist as a included matrix entry, so treat it like 'master'.
+        if ($entry['moodle-branch'] === 'master') {
+            // If the upper range does NOT exist in the matrix, then assume the user wants the master to be tested.
+            $exists = false;
+            foreach ($matrix['include'] as $row) {
+                $currentVersion = preg_replace($re, $subst, $row['moodle-branch']); // e.g. 39, 310, 311, 400, etc.
+                if ($currentVersion == $upper) {
```

Comment:
Probably worth noting here it's doing a string vs int check which is why it's a loose comparison.